src/geos_funcs_Y_d.c

```diff
@@ -0,0 +1,346 @@
+  }
+}
+static int GetY(GEOSContextHandle_t context, const GEOSGeometry* a, double* b) {
+  char typ = GEOSGeomTypeId_r(context, a);
+  if (typ != GEOS_POINT) {
+    *(double*)b = NPY_NAN;
+    return 1;
+  } else {
+    return GEOSGeomGetY_r(context, a, b);
+  }
+}
+static int GetZ(GEOSContextHandle_t context, const GEOSGeometry* a, double* b) {
+  char typ = GEOSGeomTypeId_r(context, a);
+  if (typ != GEOS_POINT) {
+    *(double*)b = NPY_NAN;
+    return 1;
+  } else {
+    return GEOSGeomGetZ_r(context, a, b);
+  }
+}
+
+#if GEOS_SINCE_3_12_0
+static int GetM(GEOSContextHandle_t context, const GEOSGeometry* a, double* b) {
+  char typ = GEOSGeomTypeId_r(context, a);
+  if (typ != GEOS_POINT) {
+    *(double*)b = NPY_NAN;
+    return 1;
+  } else {
+    return GEOSGeomGetM_r(context, a, b);
+  }
+}
+#endif
+
+static int GetPrecision(GEOSContextHandle_t context, const GEOSGeometry* a, double* b) {
+  // GEOS returns -1 on error; 0 indicates double precision; > 0 indicates a precision
+  // grid size was set for this geometry.
+  double out = GEOSGeom_getPrecision_r(context, a);
+  if (out == -1) {
+    return 0;
+  }
+  *(double*)b = out;
+  return 1;
+}
+static int MinimumClearance(GEOSContextHandle_t context, const GEOSGeometry* a, double* b) {
+  // GEOSMinimumClearance deviates from the standard pattern:
+  // - Most GEOS functions return 0 on error, 1 on success
+  // - This function returns 2 on error, 0/1 on success
+  int retcode = GEOSMinimumClearance_r(context, a, b);
+  if (retcode == 2) {
+    return 0;
+  } else {
+    return 1;
+  }
+}
+static int GEOSMinimumBoundingRadius(GEOSContextHandle_t context, const GEOSGeometry* geom, double* radius) {
+  GEOSGeometry* center = NULL;
+
+  // GEOSMinimumBoundingCircle_r computes both center and radius
+  // We only need the radius, but must clean up the center geometry
+  GEOSGeometry* ret = GEOSMinimumBoundingCircle_r(context, geom, radius, &center);
+
+  if (ret == NULL) {
+    return 0;  // Error occurred
+  }
+
+  // Clean up temporary geometries
+  GEOSGeom_destroy_r(context, center);
+  GEOSGeom_destroy_r(context, ret);
+
+  return 1;  // Success - radius is now set
+}
+
+
+/* ========================================================================
+ * CORE OPERATION LOGIC
+ * ======================================================================== */
+
+/*
+ * Core function that performs the actual GEOS operation.
+ * This is shared between both ufunc and scalar implementations to avoid code duplication.
+ *
+ * Parameters:
+ *   ctx: GEOS context handle for thread-safe operations
+ *   func: Function pointer to the specific GEOS operation to perform
+ *   geom_obj: Shapely geometry object (Python wrapper around GEOSGeometry)
+ *   result: Pointer where the computed double result will be stored
+ *
+ * Returns:
+ *   Error state code (PGERR_SUCCESS, PGERR_NOT_A_GEOMETRY, etc.)
+ */
+static char core_Y_d_operation(GEOSContextHandle_t ctx, FuncGEOS_Y_d* func,
```

Comment:
FWIW, above we are using `context` and here `ctx`. Minor but we could as well be consistent I suppose